### ISSUE TITLE
Unexpected process exit while waiting for connection details from external dart_format.

### ISSUE THREAD

**Issue by ApplicationNinja** [eggnstone/DartFormatJetbrainsPlugin#118]
Please supply any additional information here, e.g. the source code that caused the error:

```
StdErr: /Users/ninjastudio/.pub-cache/bin/dart_format: line 8: dart: command not found
Did you install the dart_format package?
Basically just execute this:<pre>dart pub global activate dart_format</pre>
```
```OS: Mac OS X
Plugin version: 2.0.23
External dart_format version: <unknown version>
```

Same Mac OS issues, have reported this before, latest plugin still not working. :(

Log file, and yes everything is installed correctly.

2024-12-09 05:29:25.575 DEBUG: OsTools()
2024-12-09 05:29:25.576 DEBUG:   IsWindows:      false (Mac OS X)
2024-12-09 05:29:25.576 DEBUG:   TempDir:        /var/folders/l2/xs_0429d2cdb_q86xh257nlw0000gn/T/
2024-12-09 05:29:25.576 DEBUG:   $HOME:          /Users/ninjastudio
2024-12-09 05:29:25.577 DEBUG:   $SHELL:         /bin/zsh
2024-12-09 05:29:25.582 DEBUG: Starting external dart_format: /bin/zsh -c "/Users/ninjastudio/.pub-cache/bin/dart_format" --web --errors-as-json --log-to-temp-file
2024-12-09 05:29:25.594 ERROR: Error-Notification: Unexpected process exit while waiting for connection details from external dart_format.
2024-12-09 05:29:25.594 ERROR:                     StdErr: /Users/ninjastudio/.pub-cache/bin/dart_format: line 8: dart: command not found|Did you install the dart_format package?|Basically just execute this:<pre>dart pub global activate dart_format</pre>

**Comment by eggnstone** [comment]
It seems that Dart is not reachable via the command line.
Did you put Dart in the PATH variable?

Please see here:
https://docs.flutter.dev/get-started/install/macos/mobile-ios#add-flutter-to-your-path

**Comment by ApplicationNinja** [comment]
Yes, as mentioned, everything is setup correctly, everything is accessible via the command line, rebooted operating system, all the latest Dart and Flutter is installed, etc...

**Comment by ApplicationNinja** [comment]
Response:

dart_format v1.2.0 (c) 2022-2024 Mark Eggenstein
Usage: dart_format [args]
    <dart file> [<dart file> ...]    Formats the specified dart file(s)
    --config=<config JSON>           Specifies the configuration
    --dry-run, -dr                   Writes output to "<original filename>.formatted.dart"
    --errors-as-json                 Writes errors as JSON to stderr
    --log-to-console                 Logs to console
    --skip-version-check             Skips version check on start-up
    --web[service]                   Starts in web service mode

**Comment by eggnstone** [comment]
OK, this looks perfect.

Maybe the shell is the problem.
Please try this:
`/bin/zsh -c "/Users/ninjastudio/.pub-cache/bin/dart_format"`

**Comment by ApplicationNinja** [comment]
Same response as above...

**Comment by eggnstone** [comment]
OK, then I have no clue where the problem may be.

The log says that the shell says that it could not find "dart", but it obviously works when you do it.

**Comment by ApplicationNinja** [comment]
It's got to be a permissions issue, Mac is super picky... How could I run this project's source code to debug and step though it to see what's going wrong, any tips / suggestions on how you debug an extension?

**Comment by eggnstone** [comment]
Good idea 👍

Which IDE were you using to test this so far?
For the plugin you need IntelliJ.
The source code is here: https://github.com/eggnstone/DartFormatJetbrainsPlugin
Then run "Run Plugin" (there is a run configuration available).

You can turn on more logging in the file Constants.kt:
```
const val DEBUG = true
const val LOG_VERBOSE = DEBUG && true
```

**Comment by ApplicationNinja** [comment]
I am using, well trying to, in IntelliJ IDE...

I'll try today if I have time to test this out ...

**Comment by ApplicationNinja** [comment]
So running from the IDE works fine, so it has to be some sort of permissions issue. :(

2024-12-09 08:30:45.661 DEBUG: OsTools()
2024-12-09 08:30:45.662 DEBUG:   IsWindows:      false (Mac OS X)
2024-12-09 08:30:45.663 DEBUG:   TempDir:        /var/folders/l2/xs_0429d2cdb_q86xh257nlw0000gn/T/
2024-12-09 08:30:45.663 DEBUG:   $HOME:          /Users/ninjastudio
2024-12-09 08:30:45.663 DEBUG:   $SHELL:         /bin/zsh
2024-12-09 08:30:45.666 DEBUG: Starting external dart_format: /bin/zsh -c "/Users/ninjastudio/.pub-cache/bin/dart_format" --web --errors-as-json --log-to-temp-file
2024-12-09 08:30:46.086 DEBUG: ExternalDartFormat.run: baseUrl:        http://127.0.0.1:7777
2024-12-09 08:30:46.087 DEBUG: ExternalDartFormat.run: currentVersion: 1.2.0
2024-12-09 08:30:46.087 DEBUG: ExternalDartFormat.run: latestVersion:  1.2.0

**Comment by eggnstone** [comment]
@ApplicationNinja Did you find out what the problem was? Any idea how to fix it?

**Comment by ApplicationNinja** [comment]
No, I haven't had the time to debug it... :(

**Comment by eggnstone** [comment]
Can you try the latest version please? 2.0.36
Now I am pretty sure that I fixed the issue :)
